src/software/new_geom/polynomial.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include <cmath>
+#include <vector>
+
+/**
+ * Polynomial is a representation of a polynomial
+ * that can calculate values at a given input
+ */
+class Polynomial
+{
+   public:
+    // Threshold for coefficient comparison.
+    // http://www.cplusplus.com/forum/beginner/95128/
+    static constexpr double EPSILON = 1e-15;
+
+    /**
+     * Construct a zero polynomial
+     */
+    explicit Polynomial();
+
+    /**
+     * Construct a polynomial from coefficients
+     * s.t. n = coeffs.size() == the degree of the polynomial
+     * and of the form coeffs[0] + coeffs[1]*x^(1)
+     * + ... + coeffs[n-1]*x^(n-1)
+     *
+     * @param coeffs coefficients of the polynomial
+     */
+    explicit Polynomial(const std::vector<double> &coeffs);
+
+    /**
+     * Construct a polynomial from coefficients
+     * s.t. n = coeffs.size() == the degree of the polynomial
```

Comment:
what if this isn't true? are you pruning?

Reply:
Forgot to remove that comment, so here the `size` would not equal the degree of the polynomial, you would need to use `getDegree`

Reply:
Addressed in 37e8554cd0bbfba2772160f2707effbccccc